lib/index.js

```diff
@@ -26,41 +26,74 @@ const noop = () => {}
 
 module.exports = getLicense
 function getLicense (modulePath, optionalCallback) {
-  const cb = optionalCallback || noop
-
   return new Promise((resolve, reject) => {
-    let pkg = {}
+    let done = finalize(resolve, reject, optionalCallback || noop)
+
+    readPackage(modulePath, (err, pkg) => {
+      if (err) return done(err)
+
+      getLicensePath(modulePath, (err, licensePath) => {
+        if (err) return done(err)
 
-    try {
-      pkg = require(path.join(modulePath, 'package.json'))
-    } catch (err) {
-      // make sure to invoke callback as well in case of an error here
+        const res = {
+          license: pkg.license,
+          licenseFile: licensePath,
+          repo: getRepo(pkg),
+          private: !!pkg.private
+        }
+
+        done(null, res)
+      })
+    })
+  })
+}
+
+// return data for both callback && promise interface
+function finalize (resolve, reject, cb) {
+  return (err, res) => {
+    if (err) {
       cb(err)
-      throw err
+      return reject(err)
     }
 
-    const tasks = LICENSES.map(makeTask.bind(this, modulePath))
+    cb(null, res)
+    resolve(res)
+  }
+}
 
-    parallel(tasks, (err, licensePath) => {
-      if (err) {
-        cb(err)
-        return reject(err)
+function readPackage (modulePath, cb) {
+  fs.readFile(path.join(modulePath, 'package.json'), 'utf8', (err, content) => {
+    let pkg = {}
+
+    if (!err) {
+      try {
+        pkg = JSON.parse(content)
+      } catch (e) {
+        // ignored
       }
+    }
+
+    // keeping the error first convention even if we never return an error here
+    cb(null, pkg)
+  })
+}
+
+function getLicensePath (modulePath, cb) {
+  fs.readdir(modulePath, (err, results) => {
+    const files = err ? LICENSES : results
```

Comment:
wouldn't it be better to handle a non-existing directory with something like:

```js
if (err) {
  const err = new Error('path does not exist')
  cb(err)
}
```

currently it takes the list of possible license files and iterates over it, even if the directory does not exist or is not readable